src/recommendations/RecommendationsPage.jsx

```diff
@@ -21,18 +22,36 @@ import RecommendationsLargeLayout from './RecommendationsPageLayouts/LargeLayout
 import RecommendationsSmallLayout from './RecommendationsPageLayouts/SmallLayout';
 import { LINK_TIMEOUT, trackRecommendationsViewed, trackSkipButtonClicked } from './track';
 import { DEFAULT_REDIRECT_URL } from '../data/constants';
+import { getAllPossibleQueryParams } from '../data/utils';
 
 const RecommendationsPage = () => {
   const { formatMessage } = useIntl();
+  const DASHBOARD_URL = getConfig().LMS_BASE_URL.concat(DEFAULT_REDIRECT_URL);
   const isExtraSmall = useMediaQuery({ maxWidth: breakpoints.extraSmall.maxWidth - 1 });
   const location = useLocation();
+  const queryParams = getAllPossibleQueryParams();
+  // flag to show recommendations for onboarding component experience
+  const showRecommendations = !!queryParams?.levelOfEducation
+    || (!!queryParams?.finalRedirectUrl && !!queryParams?.country);
+  const backendCountryCode = useSelector((state) => state.register.backendCountryCode);
 
-  const registrationResponse = location.state?.registrationResult;
-  const DASHBOARD_URL = getConfig().LMS_BASE_URL.concat(DEFAULT_REDIRECT_URL);
-  const educationLevel = EDUCATION_LEVEL_MAPPING[location.state?.educationLevel];
-  const userId = location.state?.userId;
+  const [redirectUrl, setRedirectUrl] = useState(location.state?.registrationResult?.redirectUrl);
+  const [educationLevel, setEducationLevel] = useState(EDUCATION_LEVEL_MAPPING[location.state?.educationLevel]);
+  const [userId, setUserId] = useState(location.state?.userId || -1);
+  const [userCountry, setUserCountry] = useState(backendCountryCode);
+
+  useEffect(() => {
+    if (showRecommendations) {
+      const authenticatedUser = getAuthenticatedUser();
+      if (authenticatedUser) {
+        setRedirectUrl(queryParams.finalRedirectUrl);
+        setEducationLevel(EDUCATION_LEVEL_MAPPING[queryParams.levelOfEducation]);
```

Comment:
as level of education is optional, this would crash on `queryParams.levelOfEducation`.